### PR TITLE
Update ethers rs version

### PR DESCRIPTION
**Pull request by sree-revoori1**
It is currently very hard to use this sdk and ethers rs in tandem due to the version of ethers being very outdated and types mismatching. This PR updates the ethers version to the latest version.

Also, run cargo fmt, run cargo clippy, and ignore Cargo.lock as this is a library.

### CODE REVIEW

File: src/signature/create_signature.rs
Comment (lmlmt) [inline src/signature/create_signature.rs:105]: nit: `rec_id`?

**Review by lmlmt — APPROVED** [review]
Looks good, will merge once you fix merge conflicts

File: src/signature/create_signature.rs
Comment (sree-revoori1) [reply src/signature/create_signature.rs]: Done

**Comment by sree-revoori1** [comment]
@lmlmt resolved the merge conflicts.

**Pull request merged**
